src/frequenz/sdk/_internal/_channels.py

```diff
@@ -24,3 +25,26 @@ def new_receiver(self, maxsize: int = 50) -> Receiver[T]:
         Returns:
             A receiver instance.
         """
+
+
+class LatestValueCache(typing.Generic[T]):
```

Comment:
`BackgroundService` :angry: 

It is really as easy as:

```py
class LatestValueCache(typing.Generic[T], BackgroundService):
    """A cache that stores the latest value in a receiver."""

    def __init__(self, receiver: Receiver[T]) -> None:
        """Create a new cache.
        Args:
            receiver: The receiver to cache.
        """
        self._receiver = receiver
        self._latest_value: T | None = None
        self.start()  # if you really want to start it automatically

    def start(self) -> None:
        self._tasks.add(asyncio.create_task(self._run())

    # same as before
 ```

And then you get:
* A common interface to start/stop stuff that spawn tasks
* All the async context manager goodies, stopping, etc.

Reply:
Leads to circular dependencies unfortunately.  Need to get `BackgroundService` out of the `actor` package first, I guess.